test/e2e/tests/contract-interactions.spec.js

```diff
@@ -11,64 +11,40 @@ describe('Deploy contract and call contract methods', function () {
       },
     ],
   };
+  const smartContract = 'piggybank';
   it('should display the correct account balance after contract interactions', async function () {
     await withFixtures(
       {
         dapp: true,
         fixtures: 'connected-state',
         ganacheOptions,
+        smartContract,
         title: this.test.title,
       },
-      async ({ driver }) => {
+      async ({ driver, contractRegistry }) => {
+        const contractAddress = await contractRegistry.getContractAddress(
+          smartContract,
+        );
         await driver.navigate();
         await driver.fill('#password', 'correct horse battery staple');
         await driver.press('#password', driver.Key.ENTER);
 
         // deploy contract
-        await driver.openNewPage('http://127.0.0.1:8080/');
-        await driver.clickElement('#deployButton');
+        await driver.openNewPage(
+          `http://127.0.0.1:8080/?contract=${contractAddress}`,
+        );
+
+        // wait for deployed contract, calls and confirms a contract method where ETH is sent
+        await driver.findClickableElement('#deployButton');
```

Comment:
Hmmm, I'm not sure about this as clickElement uses findClickableElement under the hood, so technically this step is redundant. I had a similar issue in #15481 and opted to use a delay instead

Reply:
So these 2 functions are pointing to different elements:
 `findClickableElement` is pointing to `#deployButton`, whereas `clickElement` is pointing to `#depositButton`.

The reason for adding this previous step of `findClickableElement` is because I noticed that until `DeployButton` element is not clickable, the Dapp is not actually ready and clicking on any other element does not work. If you look closely at the video, you'll see that this element takes slightly a bit more than the rest to be enabled.

I was considering a delay, but thought this approach is more deterministic. I could change it for a delay if we think it's clearer though. Let me know what you think @PeterYinusa 

![deploy-contract](https://user-images.githubusercontent.com/54408225/184648128-028ae5ab-91bf-4067-9516-5c6d4f983eca.gif)
![deploy-contract](https://user-images.githubusercontent.com/54408225/184648356-167aa1af-64ca-48c2-bc8f-ad25bc4c78b7.gif)

Reply:
Oops, thanks for pointing this out. This looks good as is, no change necessary